### PR TITLE
Handle authentication error messages

### PR DESCRIPTION
**Pull request by 715209**
The authentication failed and missing authentication were not handled on
the handshake, they would instead return a failed deserializing message.

image

### CODE REVIEW

**Comment by dnaka91** [comment]
Thank you for the PR! I barely use OBS without authentication, so I didn't notice this.

But, I don't really like to rely on text messages to identify the error, as the obs-websocket maintainers can change it at any time and even 1 different char and our implementation fails. Instead, I'll update `read_message` to handle the WebSocket message differently, so we can get the close code instead of always turning it into text with `into_text`.

Basically have to do something similar to how we do it already **after** the handshake: https://github.com/dnaka91/obws/blob/856ad9d0348d11d98f883878e477b8dad1fcbe94/src/client/mod.rs#L230-L238

**Comment by dnaka91** [comment]
I pushed a few changes that should now properly handle close codes in the handshake. It'll not mention these in the error message, but you can `match` on the `HandshakeError` to get out further details.

Added a `WebSocketCloseCode` enum as well, that allows to further pin down the error (if the `reason` text doesn't suffice).

**Comment by 715209** [comment]
Thanks for the quick changes, everything works as expected! 👍
